src/core/opentelemetry.ml

```diff
@@ -845,15 +919,19 @@ module Scope = struct
       call it if there is an instrumentation backend. *)
   let[@inline] add_attrs (scope : t) (attrs : unit -> key_value list) : unit =
     if Collector.has_backend () then
-      scope.attrs <- List.rev_append (attrs ()) scope.attrs
+      scope.items <-
+        List.fold_left (fun acc attr -> Attr (attr, acc)) scope.items (attrs ())
```

Comment:
I didn't think about the order; does it matter ?

Reply:
I don't think it should matter, no :)